pynest/nest/tests/test_aeif_lsodar.py

```diff
@@ -0,0 +1,246 @@
+``doc/model_details/aeif_models_implementation.ipynb``).
+
+The reference solution is stored in ``test_aeif_psc_lsodar.dat`` and was
+generated using the same dictionary of parameters, the data is then downsampled
+to keep one value every 0.01 ms and compare with the NEST simulation.
+
+The new implementation binds V_m to be smaller than 0.
+
+Rationale of the test: all models should be close to the reference LSODAR when
+submitted to the same excitatory current.
+
+Details:
+  The models are compared and we assess that the difference is smaller than a
+  given tolerance.
+"""
+
+HAVE_GSL = nest.sli_func("statusdict/have_gsl ::")
+
+
+# --------------------------------------------------------------------------- #
+#  Tolerances
+# -------------------------
+#
+
+# for the state variables (compare LSODAR and NEST implementations)
+tol_compare_V = 2e-3  # higher for V because of the divergence at spike time
+tol_compare_w = 5e-5  # better for w
+
+
+# --------------------------------------------------------------------------- #
+#  Individual dynamics
+# -------------------------
+#
+
+models = [
+  "aeif_cond_alpha",
+  # "aeif_cond_alpha_RK5",
+  "aeif_cond_exp",
+  # "aeif_cond_alpha_multisynapse"
+]
+
+num_models = len(models)
+
+# parameters with which the LSODAR reference solution was generated
+aeif_param = {
+    'V_reset': -58.,
+    'V_peak': 0.0,
+    'V_th': -50.,
+    'I_e': 420.,
+    'g_L': 11.,
+    'tau_w': 300.,
+    'E_L': -70.,
+    'Delta_T': 2.,
+    'a': 3.,
+    'b': 0.,
+    'C_m': 200.,
+    'V_m': -70.,
+    'w': 5.
+}
+
+
+# --------------------------------------------------------------------------- #
+#  Comparison function
+# -------------------------
+#
+
+def _find_idx_nearest(array, values):
+    '''
+    Find the indices of the nearest elements of `values` in `array`.
+    Both ``array`` and ``values`` should be ``numpy.array``s and `array` MUST
+    be sorted in increasing order.
+
+    Parameters
+    ----------
+    array : reference list or np.ndarray
+    values : double, list or array of values to find in `array`
+
+    Returns
+    -------
+    idx : int or array representing the index of the closest value in `array`
+    '''
+    idx = np.searchsorted(array, values, side="left")  # get the interval
+    # return the index of the closest
+    if isinstance(values, float) or isinstance(values, int):
+        if idx == len(array):
+            return idx - 1
+        else:
+            return idx - (np.abs(values - array[idx - 1]) <
+                          np.abs(values - array[idx]))
+    else:
+        # find where it is idx_max+1
+        overflow = (idx == len(array))
+        idx[overflow] -= 1
+        # for the others, find the nearest
+        tmp = idx[~overflow]
+        idx[~overflow] = tmp - (np.abs(values[~overflow] - array[tmp - 1]) <
+                                np.abs(values[~overflow] - array[tmp]))
+        return idx
+
+
+def _interpolate_lsodar(nest_times, lsodar):
```

Comment:
Because when the problem occurred I did not think about it and I did it on the fly, which is pretty stupid... I'll have a look at them.